### ISSUE TITLE
Swift version

### ISSUE THREAD

**Issue by eantonopoulou** [jumartin/Calendar#29]
Hello!Your calendar is great!I ;m trying to convert it to Swift and a'i m having some trouble there.Will a Swift compatible version be available anytime soon?Or where should i start??Thank you in advance!

**Comment by jumartin** [comment]
Hi,
I had no problem using the lib with Swift... 
What kind of trouble did you get?

**Comment by eantonopoulou** [comment]
Hello again!Thank you for your quick response!
Swift is working just fine!I'm just a little bit new at this!Could you give me an example on how to use the module without the use of EventKit?

Thank you

**Comment by phunagewine** [comment]
Pretty much what @eantonopoulou requested, I tried to not use EKEventViewController, but due to its dependency tied to the calendar, I find myself reinventing a custom view controller to follow the event data source method. Not much luck heading in this direction, wish to have some tips and quick direction on this @jumartin 
Thanks for the wonderful calendar control!

**Comment by eantonopoulou** [comment]
Yes, exactly like @phunagewine !

**Comment by eantonopoulou** [comment]
Hello again!I managed to create custom controllers without the use of EventKit.The only problem now is that the scrolling in DayView stops at 23:00.Is this normal behavior?

**Comment by eantonopoulou** [comment]
My bad, i was doing something wrong with the view bounds!Works fine.

**Comment by ReiVerdugo** [comment]
Hello. Does someone have an example swift project? I am kinda lost here u__u

**Comment by eantonopoulou** [comment]
Hello! I have attached the source files from my first attempt to convert the example project in Swift.It's a little bit slappy, but it might help you!
[TestCalendarLibForSwift.zip](https://github.com/jumartin/Calendar/files/371341/TestCalendarLibForSwift.zip)

**Comment by ReiVerdugo** [comment]
@eantonopoulou Tysm!

**Comment by YogeshBharate** [comment]
@eantonopoulou  @jumartin  :

I am using your library in my iPad application. I am using Swift 2.2. How to add events in week view ? Thanks in advance.

**Comment by eantonopoulou** [comment]
Hello,

I haven't used the add event functionality but as i see it you have to set the dayPlannerView.canCreateEvents  to  true and handle the events:
func dayPlannerView(view: MGCDayPlannerView!, canCreateNewEventOfType type: MGCEventType, atDate date: NSDate!) -> Bool
func dayPlannerView(view: MGCDayPlannerView!, createNewEventOfType type: MGCEventType, atDate date: NSDate!)

Not sure though!

**Comment by tejas-ardeshna** [comment]
Please release swift version. I am facing many issue with objc to swift.

**Comment by bhadjajay23** [comment]
Hello,
create event , move event n edit event not working in swift. Please release proper swift version.

**Comment by NikunjGodhani** [comment]
Hi eantonopoulou ,
I am searching swift version of this calendar. If you have Full version of the Swift than Please Send me on godhani_n58@yahoo.in

Plz any user have converted swift version of this calendar than please on this email address.

**Comment by duraiganesh0** [comment]
Yes. i am also looking for a swift tutorial to use this library. Can anyone help me with this? @jumartin @eantonopoulou

**Comment by patel5524** [comment]
Yes. i am also looking for a swift tutorial to use this library.i can not new event create.Can anyone help me with this? @jumartin @eantonopoulou

**Comment by haketputek** [comment]
I would also want a swift version of this. Thanks!

**Comment by florianhenonin** [comment]
Yes, me too. Thanks

**Comment by Nefelium** [comment]
Please, give me too. I need this calendar on Swift. Nefelium@gmail.com
Thanx.
